### PR TITLE
O mode

### PR DESCRIPTION
**Pull request by bettinson**
Added vim-like o-mode to insert a line above ('O') and below ('o') the cursor in normal mode.

### CODE REVIEW

File: src/process_keypress.c [hunk 1]
```diff
@@ -69,11 +69,20 @@ void editorProcessKeypress(int fd) {
     case 'l':
       editorMoveCursor(RIGHT);
       break;
+	case 'o':
+	  editorInsertRow(E.cy + E.rowoff + 1, "\n", 0);
+      editorMoveCursor(DOWN);
+      mode = VM_INSERT;
```
Comment (jpittis) [inline src/process_keypress.c:75]: this should be tabbed over.

File: src/process_keypress.c [hunk 2]
```diff
@@ -69,11 +69,16 @@ void editorProcessKeypress(int fd) {
     case 'l':
       editorMoveCursor(RIGHT);
       break;
+	case 'o':
+	  editorInsertRow(E.cy + E.rowoff + 1, "\n", 0);
```
Comment (epilk) [inline src/process_keypress.c]: s/0/1/

File: src/process_keypress.c [hunk 3]
```diff
@@ -69,11 +69,16 @@ void editorProcessKeypress(int fd) {
...
+      editorMoveCursor(DOWN);
+      mode = VM_INSERT;
+	  break;
```
Comment (epilk) [inline src/process_keypress.c]: align

File: src/process_keypress.c [hunk 4]
```diff
@@ -69,11 +69,16 @@ void editorProcessKeypress(int fd) {
...
     case 'q':
       exit(0);
     case 'i':
-      mode = VM_INSERT;
       editorSetStatusMessage("INSERT");
+      mode = VM_INSERT;
```
Comment (epilk) [inline src/process_keypress.c:85]: Thanks for this quality change

File: src/process_keypress.c [hunk 5]
```diff
@@ -74,6 +74,10 @@ void editorProcessKeypress(int fd) {
       editorMoveCursor(DOWN);
       mode = VM_INSERT;
 	  break;
+	case 'O':
+	  editorInsertRow(E.cy + E.rowoff, "\n", 0);
```
Comment (epilk) [inline src/process_keypress.c]: ditto

**Comment by jpittis** [comment]
add this to your vimrc matt: https://github.com/jpittis/dotfiles/blob/master/.vimrc#L94

**Pull request merged**